pkg/pool-gyro/contracts/Gyro2CLPPool.sol

```diff
@@ -90,16 +90,24 @@ contract Gyro2CLPPool is IGyro2CLPPool, BalancerPoolToken {
         // New invariant
         invariant = invariant.mulUp(invariantRatio);
         uint256 squareNewInv = invariant * invariant;
-        // L / sqrt(beta)
-        uint256 a = invariant.divDown(sqrtBeta);
-        // L * sqrt(alpha)
-        uint256 b = invariant.mulDown(sqrtAlpha);
 
         if (tokenInIndex == 0) {
             // if newBalance = newX
+
+            // L / sqrt(beta), rounded down to minimize newBalance.
+            uint256 a = invariant.divDown(sqrtBeta);
+            // L * sqrt(alpha), rounded up to minimize newBalance (b is in the denominator).
+            uint256 b = invariant.mulUp(sqrtAlpha);
+
             newBalance = squareNewInv.divUpRaw(b + balancesLiveScaled18[1]) - a;
         } else {
             // if newBalance = newY
+
+            // L / sqrt(beta), rounded up to minimize newBalance (a is in the denominator).
+            uint256 a = invariant.divUp(sqrtBeta);
+            // L * sqrt(alpha), rounded down to minimize newBalance.
+            uint256 b = invariant.mulDown(sqrtAlpha);
```

Comment:
Same as above; shouldn't `a` be rounded down? I think I agree more with the original code before these changes. Am I missing something?

Reply:
note that we want `newBalance` to be as high as possible (see comments in `IBasePool`)